restful.go

```diff
@@ -56,14 +56,14 @@ type snowflakeRestful struct {
 	HeartBeat   *heartbeat
 
 	Connection          *snowflakeConn
-	FuncPostQuery       func(context.Context, *snowflakeRestful, *url.Values, map[string]string, []byte, time.Duration, *uuid.UUID) (*execResponse, error)
-	FuncPostQueryHelper func(context.Context, *snowflakeRestful, *url.Values, map[string]string, []byte, time.Duration, *uuid.UUID) (*execResponse, error)
+	FuncPostQuery       func(context.Context, *snowflakeRestful, *url.Values, map[string]string, []byte, time.Duration, string) (*execResponse, error)
+	FuncPostQueryHelper func(context.Context, *snowflakeRestful, *url.Values, map[string]string, []byte, time.Duration, string) (*execResponse, error)
 	FuncPost            func(context.Context, *snowflakeRestful, *url.URL, map[string]string, []byte, time.Duration, bool) (*http.Response, error)
 	FuncGet             func(context.Context, *snowflakeRestful, *url.URL, map[string]string, time.Duration) (*http.Response, error)
 	FuncRenewSession    func(context.Context, *snowflakeRestful, time.Duration) error
 	FuncPostAuth        func(context.Context, *snowflakeRestful, *url.Values, map[string]string, []byte, time.Duration) (*authResponse, error)
 	FuncCloseSession    func(context.Context, *snowflakeRestful, time.Duration) error
-	FuncCancelQuery     func(context.Context, *snowflakeRestful, *uuid.UUID, time.Duration) error
+	FuncCancelQuery     func(context.Context, *snowflakeRestful, string, time.Duration) error
```

Comment:
Now that the request ID lives in the context, you can probably just get rid of the request ID argument and rely only on the context object for getting the request ID. Otherwise it's unclear which to use -- getRequestID() or the argument?

Reply:
Internally, the functions that have `requestID` explicitly as an argument all use that `requestID` instead of grabbing the request ID from the context.

The three function that take `requestID` explicitly are `FuncPostQuery/FuncPostQueryHelper/FuncCancel`
The FuncPost* need the request ID because internally they have some fancy re-try logic that keeps the request ID stable across re-try, but generates random request ID for renewal and cancel.

To keep the existing behavior intact when `SNOWFLAKE_REQUEST_ID` is not set in the context, it's still simpler to keep the current argument intact and not stamp a request ID into the context if not specified.